lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -535,6 +535,146 @@ class ScaledBlockedToMMAv2
   }
 };
 
+static void updateValueType(Value v, Attribute encoding,
+                            ArrayRef<int64_t> shape) {
+  auto tensorType = cast<RankedTensorType>(v.getType());
+  auto newType =
+      RankedTensorType::get(shape, tensorType.getElementType(), encoding);
+  v.setType(newType);
+}
+
+static TransOp updateUsers(Value result, const SetVector<Operation *> &slice) {
+  TransOp transOp;
+  if (llvm::any_of(result.getUsers(),
+                   [&](Operation *user) { return slice.count(user) == 0; })) {
+    OpBuilder builder(result.getContext());
+    builder.setInsertionPointAfterValue(result);
+    transOp =
+        builder.create<TransOp>(result.getLoc(), result, ArrayRef({1, 0}));
+    result.replaceUsesWithIf(transOp.getResult(), [&](OpOperand &operand) {
+      return operand.getOwner() != transOp.getOperation() &&
```

Comment:
Why wouldn't we want to replace it in this case?

Reply:
Perhaps a better question was "in what case would we hit this"? Is this to avoid applying the same transformation twice?

Reply:
It will always be hit since the newly created `transOp` that we're replacing `result` with is itself a user of `result`. So it would generate invalid IR.